src/clock.rs

```diff
@@ -0,0 +1,28 @@
+// Copyright 2021  The Hypatia Authors
+// All rights reserved
+//
+// Use of this source code is governed by an MIT-style
+// license that can be found in the LICENSE file or at
+// https://opensource.org/licenses/MIT.
+
+use crate::cpuid;
+
+pub const NANOS_PER_SEC: u128 = 1_000_000_000;
+
+/// Returns the clock frequency of the current CPU in Hertz.
```

Comment:
I'm pretty sure this is always going to just return `DEFAULT_HZ` on AMD since leaf 15 isn't supported. I don't think it's unreasonable to just use 2GHz as a rough approximation - it's what we do in stlouis until the TSC is calibrated.

```
CPUID 0000000f:00 = 00000000 000000ff 00000000 00000002 | ................
```

Reply:
Sigh. That's a bummer. It doesn't appear to support `MSR_PLATFORM_INFO`, either; trying to read that generates a GPF. It seems like one could calculate the TSC frequency by comparing against the PTSC, which may be slightly more accurate, but I think we're probably close enough for right now?